src/main.rs

```diff
@@ -42,15 +49,139 @@ fn read_can(mut publisher: MqttClient, can_interface: &str) {
         );
         let decoded_data = message.decode();
         for data in decoded_data.iter() {
-            publisher.publish(data)
+            let mut payload = serverdata::ServerData::new();
+            payload.unit = data.unit.to_string();
+            payload.value = data.value.iter().map(|x| x.to_string()).collect();
+
+            client
+                .publish(
+                    data.topic.to_string(),
+                    protobuf::Message::write_to_bytes(&payload).unwrap_or_else(|e| {
+                        format!("failed to serialize {}", e).as_bytes().to_vec()
+                    }),
+                )
+                .expect("Could not publish!");
+            // TODO: investigate disabling this
+            thread::sleep(Duration::from_micros(100));
         }
+    });
+    Ok(join_handle)
+}
+
+/**
+ * Reads the mqtt incoming messages and sends can messages based off of that
+ */
+fn read_siren(
+    pub_path: &str,
+    send_map: Arc<RwLock<HashMap<u32, EncodeData>>>,
+) -> Result<JoinHandle<()>, String> {
+    let mut client = MqttClient::new(pub_path, "calypso-encoder");
+    client.connect().expect("Could not connect to Siren!");
+    let rx = client.start_consumer().expect("Could not begin consuming");
```

Comment:
rx is a bad variable name, most CS people dont know the difference between tx and rx

Reply:
i dont agree with peyton but i also dont care